03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,338 @@
+objects to your environment.
+
+The data that we downloaded from refine.bio for this analysis is a dataset, with
+a total of 24 samples (obtained from 6 mice with 4 technical replicates for each
+mouse), containing RNA-sequencing results for Group 3 and 4 medulloblastoma
+orthotopic xenograft mice with and without digoxin treatment.
+
+You can download this dataset by going to the following link and clicking 
+`Download Now`: https://www.refine.bio/experiments/SRP150101/transcriptome-analysis-of-group-3-and-4-medulloblastoma-orthotopic-xenograft-mice-with-digoxin-treatment
+
+```{r Import data from .tsv files}
+# Read in metadata tsv file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP150101.tsv" # Replace with the name of your metadata file
+)) %>%
+  dplyr::select(refinebio_accession_code, refinebio_title, refinebio_treatment) %>%
+  replace(is.na(.), "None") %>%
+  dplyr::mutate(
+    groups = dplyr::case_when(
+    refinebio_treatment == "no treatment" ~ "control",
+    TRUE ~ "digoxin"
+    # Replace with group names relevant to your data
+    ),
+    # Create a variable to hold the sample summary information
+    # (excluding the lane information)
+    sample_info_summary = gsub("(_L).*", "", refinebio_title),
+    # Create a variable to store the mouse ids
+    mouse_id = stringr::word(refinebio_title, sep = "-|_", 2, 3),
+    # Create a variable to store the subgroup information
+    subgroup = dplyr::case_when(
+      startsWith(mouse_id, "2555MB") ~ "group 3",
+      startsWith(mouse_id, "1078MB") ~ "group 4",
+      TRUE ~ "unknown"
+  ))
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP150101.tsv" # Replace with the name of your data file
+)) %>%
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## 3) Choose genes of interest
+Although you may want to create a heatmap including all of the genes in the set,
+alternatively, the heatmap could be created using only genes of interest. 
+For this example, we will sort genes by variance, but there are many alternative
+criterion by which you may want to sort your genes <i>eg</i> fold change,
+t-statistic, membership to a particular gene ontology, so on. 
+
+```{r Choose genes}
+# Calculate the variance for each gene
+variances <- apply(df, 1, var)
+
+# Determine the upper quartile variance cutoff value
+upper_var <- quantile(variances, 0.75)
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df_by_var <- df %>%
+  dplyr::filter(variances > upper_var)
+```
+
+## How are our technical replicates clustering together?
+
```

Comment:
This is great to start out with the question like we discussed. Can you try to add some words here to explain why we are doing this and give a brief summary of what we will be doing?